src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -726,7 +727,25 @@ namespace xclhwemhal2 {
     unsetenv("VITIS_KERNEL_PROFILE_FILENAME");
     unsetenv("VITIS_KERNEL_TRACE_FILENAME");
 
-    if (!simDontRun)
+    if (args.m_emuData) {
+      extractEmuData(binaryDirectory, binaryCounter, args);
+      std::string emuSettingsFilePath = binaryDirectory + "/emulation_data/emu_meta_data.json";
+      if (boost::filesystem::exists(emuSettingsFilePath)) {
+        if (readEmuSettingsJsonFile(emuSettingsFilePath)) {
+          if (simDontRun) {
+            mSimDontRun = simDontRun;
+          }
+        }
+      }
+      else {
+        mSimDontRun = simDontRun;
+      }
+    }
+    else {
+      mSimDontRun = simDontRun;
+    }
+
```

Comment:
As discussed, Please simplify this code.

Reply:
committed the change as per the review comments